conda/shell/etc/fish/conf.d/conda.fish

```diff
@@ -72,7 +72,11 @@ function conda --inherit-variable CONDA_EXE
                 eval ($CONDA_EXE shell.fish $cmd $argv)
             case install update upgrade remove uninstall
                 $CONDA_EXE $cmd $argv
-                and eval ($CONDA_EXE shell.fish reactivate)
+                set -l cmd_status $status
+                if test $cmd_status -eq 0
+                    eval ($CONDA_EXE shell.fish reactivate)
+                end
+                return $cmd_status
```

Comment:
I think we still want the return code of reactivate if its run, so either of the following:

```suggestion
                set -l cmd_status $status
                if test $cmd_status -eq 0
                    eval ($CONDA_EXE shell.fish reactivate)
                    return $status
                end
                return $cmd_status
```

```suggestion
                set -l cmd_status $status
                if test $cmd_status -ne 0
                    return $cmd_status
                end
                eval ($CONDA_EXE shell.fish reactivate)
```

Reply:
Your suggested changes are functionally equivalent to the one-line version before my suggested commit. So are you saying it is intended behavior that `conda update` fails with an exit code of 4 in the fish shell when e.g. the environment variable _CE_M is not set? I don’t think so. A correct, real alternative is if instead the fish code returned by `reactivate` is changed so that this situation does not cause an error. Would this be preferred?